doc/source/WorkingWithTheNodeGraph/Contexts/index.md

```diff
@@ -0,0 +1,276 @@
+
+- Frame number
+- Scene location
+- Image tile
+- Image channel
+
+Gaffer automatically manages Contexts during processing. These are silently passed upstream, from node to node as they compute requested data. Certain nodes can also add and modify their own variables, for use in driving graph behaviour.
+
+There is no single interface in Gaffer for manually controlling and inspecting a Context in its entirety, but nodes like NameSwitch, Spreadsheet, and Expression can add Context-dependent behaviour to graphs, and the [Python API](../../WorkingWithThePythonScriptingAPI/index.md) provides methods for creating and inspecting Contexts.
+
+
+## Nodes and Contexts ##
+
+In order to provide a background for why Contexts are necessary, we should look at what kind of workload a node would be up against without them.
+
+Let's consider a 3D scene. Keeping in mind [how scenes are structured](../../WorkingWithScenes/AnatomyOfAScene/index.md), imagine it is filled with hundreds of thousands of locations. Clearly, not every location can be processed at once. Gaffer needs a way to divide up the work to compute the locations one at a time.
+
+This is where Contexts come in. In Gaffer, at no point does a node operate on the entire scene/image. Contexts break the scene/image down into digestible fragments, so each node only generates the parts that they were told to, while deferring the rest. Each computing thread has its own Context, so multiple threads may be processing different parts of the scene/image in parallel, greatly improving efficiency and performance.
+
+```eval_rst
+.. figure:: images/illustrationContextsSlices.png
+    :scale: 100%
+    :alt: A scene file being sliced up by location.
+    
+    **Figure A:** Contexts allow a scene file to be processed in slices, one location at a time.
+```
+
+
+## Context flow ##
+
+While scene and image data flows down the graph, Contexts flow upstream.
+
+Computation begins when a node is queried to compute a value for its output plug, by something such as a dispatcher or an editor in the interface. At the start of the computation, whatever made the request creates a Context and sets its variables to specify the target location, set, image tile, channel, etc.
+
+The Context then passes up to any plugs that contribute to the output value. When an input plug on the queried node is connected to a plug on an upstream node, it pulls on that upstream plug to retrieve its value. If necessary, the upstream node may need to compute in turn, to produce the upstream plug's value. This pull-compute reaction occurs up through the graph until all required plug values have been calculated to create an output value for the original queried node.
+
+```eval_rst
+.. figure:: images/illustrationContextsFlow.png
```

Comment:
![image](https://user-images.githubusercontent.com/896779/81818541-11506a00-9526-11ea-9926-e7be6b9232e6.png)
Seems to be broken in the CI linux build

Reply:
This image is still broken in https://github.com/GafferHQ/gaffer/commit/953a60d13f22d3ed79f342c40ef7a4f8e192bb02

Reply:
This image is still broken in https://github.com/GafferHQ/gaffer/commit/dc7d7e2a2f7c7f1640c4f5083518ff3dfe44537e. See the linux CI build: https://gafferhq.blob.core.windows.net/builds/gaffer-pr3721-docsContext-2020_05_15_2307-dc7d7e2a-linux.tar.gz